### PR TITLE
test(gemma4): the checkpoint gates can only be produced for a checkpoint that fits beside its own reference on one card

### LINKED ISSUE #1049: gemma4: the checkpoint gates can only be produced for a checkpoint that fits beside its own reference on one card
## Description

The checkpoint gates read their four HF fixtures and the chat reference from fixed paths under `test_data/`, and the dumpers load the reference tower on one device. Both are 12B by construction: another checkpoint's fixtures have nowhere to live under the same names, and a checkpoint whose reference tower does not fit on one card next to the long-context prompts cannot be dumped at all. M3 (#986) needs the same suite at 31B.

## Proposed Plan

- The gates take their fixture paths from the environment, with the committed 12B files as the defaults.
- The runner selects a fixture set by tag (`PEGAINFER_GEMMA4_FIXTURE_TAG`, default `12b`), exports the paths, and holds the set against the checkpoint's digests before the first load, as it does today.
- The dumpers can shard the reference tower over several GPUs; the golden dumper fingerprints a sharded checkpoint the same way it fingerprints a single file; the chat parity reference follows the tag.

## Acceptance Criteria

- The committed 12B suite runs unchanged with nothing set.
- A 31B set dumped under the tag runs the whole suite through the runner with only the tag and the checkpoint paths set.

### PR DESCRIPTION
**Pull request by FeathBow**
## Description

Closes #1049

**The suite was 12B by construction.** The checkpoint gates opened their four HF fixtures and the chat reference at fixed paths, and the dumpers loaded the reference tower on one device. Another checkpoint's fixtures had nowhere to live, and a tower that does not fit beside the long-context prompts on one card could not be dumped.

**Paths from the environment, a set by tag.** Each gate reads its fixture path from `PEGAINFER_GEMMA4_{GOLDEN,WINDOW_GOLDEN,LONGCTX_GOLDEN,GENERATE,CHAT_GOLDEN}`, falling back to the committed 12B files. `scripts/gemma4_gates.sh` takes `PEGAINFER_GEMMA4_FIXTURE_TAG` (default `12b`; the committed chat reference keeps its untagged name), exports the five paths for that tag, and holds the set against the checkpoint's digests before the first load, as before.

**Dumped sharded when the tower needs it.** The dumpers accept a device map that shards the reference tower over several GPUs; the golden dumper fingerprints a sharded checkpoint by its index and shard headers the way it fingerprints a single file; the chat parity reference follows the tag.

## Test Env

- Single GPU (sm_89, x86_64) for the 12B suite; one and two GH200s (sm_90, aarch64) for the 31B dumps and suite.

## Verification

Bound to the tip of this range.

- The 12B suite through the runner with nothing set: every gate resolves to the committed files and passes as before.
- The 31B set dumped from `google/gemma-4-31B-it` at revision `842da3794eaa` (window and long-context cases with the tower sharded over two GH200s) and the whole suite run through the runner with `PEGAINFER_GEMMA4_FIXTURE_TAG=31b`: the fixture preflight agrees on the revision, and the suite runs to its end; the numeric results are retained in the M3 snapshot rather than here.
- `cargo fmt --check`, `clippy -D warnings` on the gemma4 and frontend crates including tests.

### CODE REVIEW

**Comment by FeathBow** [comment]
@codex review please

**Comment by FeathBow** [comment]
@codex review thx

File: scripts/gemma4_gates.sh [hunk 1]
```diff
@@ -132,14 +132,24 @@ GATES_GEMMA4_TOKENIZER_PARITY=(
   "ckpt,chatgolden string_form_chat_renders_match_hf_reference"
 )
 
-CHAT_GOLDEN=test_data/gemma4-tokenizer-golden.json
+# The fixture set is named by the checkpoint it was dumped from; the
+# committed set is 12b. Another tag selects fixtures dumped for another
+# checkpoint under the same names, and the gates are told where they are.
+FIXTURE_TAG=${PEGAINFER_GEMMA4_FIXTURE_TAG:-12b}
```
[the remark this answers is not in the document]
Comment (FeathBow) [reply scripts/gemma4_gates.sh:138]: Makes sense

File: tools/accuracy/dump_gemma4_hf_golden.py [hunk 2]
```diff
@@ -241,9 +259,11 @@ def main() -> int:
     tensors: dict[str, torch.Tensor] = {}
     for name, tokens, probed in cases:
         case_cuts = cuts if probed else []
-        hidden, logits = run_case(model, text_model, case_cuts, tokens, args.device)
+        hidden, logits = run_case(
+            model, text_model, case_cuts, tokens, input_device(model, args.device)
```
[the remark this answers is not in the document]
Comment (FeathBow) [reply tools/accuracy/dump_gemma4_hf_golden.py:265]: Sure
